files/routes/admin/admin.py

```diff
@@ -277,29 +281,48 @@ def update_filter_status(v):
 	if new_status not in ['normal', 'removed', 'ignored']:
 		return { 'result': f'Status of {new_status} is not permitted' }
 
+	if new_status == 'normal':
+		state_mod_new = StateMod.Visible
+		state_report_new = StateReport.Resolved
+	elif new_status == 'removed':
+		state_mod_new = StateMod.Removed
+		state_report_new = StateReport.Resolved
+	elif new_status == 'ignored':
+		state_mod_new = None	# we just leave this as-is
+		state_report_new = StateReport.Ignored
+
 	if post_id:
 		target = g.db.get(Submission, post_id)
-		old_status = target.filter_state
-		rows_updated = g.db.query(Submission).where(Submission.id == post_id) \
-							.update({Submission.filter_state: new_status})
+		old_status = target.state_mod
+
+		# this could totally be one query but it would be kinda ugly
+		if state_mod_new:
```

Comment:
Question: if state_mod_new = StateMod.Visible (which has the integer value of 0), will this work?

The answer might surprise you, so to avoid double takes etc I suggest making StateMod and StateReport start from 1 (see also: https://docs.python.org/3/library/enum.html#enum.Enum._generate_next_value_ and all the examples there).

Reply:
Honestly I just should've used `is not None` here. My mistake.

Reply:
I still think that fixing the issue at the source (that is, not using enum values of 0) is a better approach. Your choice ofc, it works as it is even.